internal/entities/service/service.go

```diff
@@ -0,0 +1,332 @@
+// SPDX-FileCopyrightText: Copyright 2024 The Minder Authors
+// SPDX-License-Identifier: Apache-2.0
+
+// Package service contains the service layer for entity instances
+package service
+
+import (
+	"context"
+	"database/sql"
+	"errors"
+	"fmt"
+
+	"github.com/google/uuid"
+	"github.com/rs/zerolog"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"github.com/mindersec/minder/internal/db"
+	"github.com/mindersec/minder/internal/engine/entities"
+	"github.com/mindersec/minder/internal/entities/models"
+	propService "github.com/mindersec/minder/internal/entities/properties/service"
+	"github.com/mindersec/minder/internal/providers/manager"
+	"github.com/mindersec/minder/internal/util"
+	pb "github.com/mindersec/minder/pkg/api/protobuf/go/minder/v1"
+)
+
+//go:generate go run go.uber.org/mock/mockgen -package mock_$GOPACKAGE -destination=./mock/$GOFILE -source=./$GOFILE
+
+// ErrEntityNotFound is returned when an entity is not found
+var ErrEntityNotFound = errors.New("entity not found")
+
+// EntityService encapsulates logic related to entity instances
+type EntityService interface {
```

Comment:
We're relying on GPRC implementing the "unimplemented" result for RegisterEntity, correct?

Reply:
duh! yes, we could do that. I left the implementation and returned an error explicitly. Let me remove that entirely.